integrations/helm/release/release.go

```diff
@@ -296,3 +298,49 @@ func mergeValues(dest, src chartutil.Values) chartutil.Values {
 	}
 	return dest
 }
+
+func releaseManifestToUnstructured(manifest string, logger log.Logger) []unstructured.Unstructured {
+	manifests := helmutil.SplitManifests(manifest)
+	var objs []unstructured.Unstructured
+	for _, manifest := range manifests {
+		json, err := yaml.YAMLToJSON([]byte(manifest))
+		if err != nil {
+			logger.Log("err", err)
+			continue
+		}
+
+		var u unstructured.Unstructured
+		if err := u.UnmarshalJSON(json); err != nil {
+			logger.Log("err", err)
+			continue
+		}
+
+		// Helm charts may include list kinds, we are only interested in
+		// the items on those lists.
+		if u.IsList() {
+			l, err := u.ToList()
+			if err != nil {
+				logger.Log("err", err)
+				continue
+			}
+			objs = append(objs, l.Items...)
+			continue
+		}
+
+		objs = append(objs, u)
+	}
+	return objs
+}
+
+func namespacedResourceMap(objs []unstructured.Unstructured, defaultNamespace string) map[string][]string {
+	resources := make(map[string][]string)
+	for _, obj := range objs {
+		namespace := obj.GetNamespace()
+		if namespace == "" {
+			namespace = defaultNamespace
```

Comment:
The name `defaultNamespace` misled me into thinking this code was fallible, since it had to be given a hard-wired default namespace. But it's OK in fact, because it is the release namespace that is passed, which is indeed the namespace given to anything that needs a namespace and hasn't got one. Would calling the argument `releaseNamespace` make this clearer do you think, or am I just obsessed with default namespaces after #1442?

Reply:
It misled me too now that I hadn't seen this code in a couple of days, which is a sign. Will adjust it.